everestrs/everestrs/src/lib.rs

```diff
@@ -38,6 +39,52 @@ mod ffi {
         data: Vec<u8>,
     }
 
+    /// The possible types a config can have. Note: Naturally this would be am
+    /// enum **with** values - however, cxx can't (for now) map Rusts enums to
+    /// std::variant or union.
+    enum ConfigType {
```

Comment:
use `serde_codegen::schema::interface::Value` and use Serde do deserialize stuff. That makes this code go away and the other code can be reused.

Reply:
This means pulling this into everstrs which seems fine.

Reply:
Are you suggesting to serialize the stuff in the c++ land and then pass json to rust and then deserialize it again?

Reply:
actually I misunderstood the C++ stuff - I was assuming that json values are passed around, but for Config that is not the case. retracted.